frontend/src/views/TE/Upload.vue

```diff
@@ -190,6 +194,65 @@ export default class Upload extends Vue {
     }
   }
 
+  getFileHash(file: File) {
+    return new Promise(resolve => {
+      const fileReader = new FileReader();
+      fileReader.onload = function(event: any) {
+        const spark = new Spark.ArrayBuffer();
+        spark.append(event.target.result);
+        resolve(btoa(spark.end(true)));
+      };
+      fileReader.readAsArrayBuffer(file);
+    });
+  }
+
+  async validateFile(file: File) {
+    if (file.type !== "audio/flac") {
+      throw new Error("File type must be flac");
+    }
+    if (!file.name.match(flacFileFormat)) {
+      throw new Error("The file name is not a correct task ID");
+    }
+    const taskId = getTaskId(file.name);
+    const task = (
+      await firebase
+        .database()
+        .ref(`/TE/tasks/${taskId}`)
+        .once("value")
+    ).val();
+    if (task.status === "Done") {
```

Comment:
What if the task does not exist or not accessible to the user? I guess you missed “The task {{taskId}} is not assigned to you.” check here.

Reply:
When the task is not accessible firebase will throw a permission denied error which is displayed. Should this be detected and replaced with a different message?

Reply:
Depends on the message that Firebase throws. Can you show it to me?